storage/aptosdb/src/state_store/mod.rs

```diff
@@ -204,13 +204,20 @@ impl StateStore {
         Ok(result)
     }
 
-    /// Read the value for a specific version and require its existence. Call `get_value_by_value`
-    /// for most use cases.
-    fn get_value_at_version(&self, key_and_version: &(StateKey, Version)) -> Result<StateValue> {
-        self.ledger_db
-            .get::<StateValueSchema>(key_and_version)
-            .and_then(|v| {
-                v.ok_or_else(|| format_err!("State Value is missing for {:?}", *key_and_version))
+    fn get_existing_value_by_version(
+        &self,
+        state_key: &StateKey,
+        version: Version,
+    ) -> Result<StateValue> {
+        self.get_value_by_version(state_key, version)
```

Comment:
Can you clarify what is the motivation behind this change? My understanding is that `get_value_by_version` will be slower than `db.get` because of the seek - or is that not the case anymore?

Reply:
@grao1991 is benchmarking this and the preliminary result is the `seek` is a little bit slower than `get` only but much faster than `get_with_proof` so we can sync tmr about the details.

We definitely need the `seek` version because lazy commit depends on this and we haven't found any workaround. If `prefix seek` can be tuned to be comparable to `get` we are good to go this route.